src/PowerShell/Microsoft.WinGet.Client/PSObjects/FoundCatalogPackage.cs

```diff
@@ -0,0 +1,89 @@
+﻿// -----------------------------------------------------------------------------
+// <copyright file="FoundCatalogPackage.cs" company="Microsoft Corporation">
+//     Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+// </copyright>
+// -----------------------------------------------------------------------------
+
+namespace Microsoft.WinGet.Client.PSObjects
+{
+    using System.Linq;
+
+    /// <summary>
+    /// FoundCatalogPackage wrapper object for displaying to PowerShell.
+    /// </summary>
+    public class FoundCatalogPackage
+    {
+        private readonly string id;
+        private readonly string name;
+        private readonly string version;
+        private readonly bool isUpdateAvailable;
+        private readonly string source;
+        private readonly string[] availableVersions;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FoundCatalogPackage"/> class.
+        /// </summary>
+        /// <param name="catalogPackage">The catalog package COM object.</param>
+        public FoundCatalogPackage(Management.Deployment.CatalogPackage catalogPackage)
+        {
+            this.id = catalogPackage.Id;
+            this.name = catalogPackage.Name;
+            this.isUpdateAvailable = catalogPackage.IsUpdateAvailable;
+            this.version = catalogPackage.DefaultInstallVersion.Version;
+            this.source = catalogPackage.DefaultInstallVersion.PackageCatalog.Info.Name;
+
+            if (this.isUpdateAvailable)
+            {
+                this.availableVersions = catalogPackage.AvailableVersions.Select(i => i.Version).ToArray();
+            }
+        }
+
+        /// <summary>
+        /// Gets the id of the catalog package.
+        /// </summary>
+        public string Id
+        {
+            get { return this.id; }
+        }
+
+        /// <summary>
+        /// Gets the name of the catalog package.
+        /// </summary>
+        public string Name
+        {
+            get { return this.name; }
+        }

```

Comment:
Instead of having a `name` field and this property, you can have just `public string Name { get; private set; }` to let C# use the default implementation. Or if you are only setting the value in the constructor, I think you can just leave it as `public string Name { get; }`

Reply:
Changed to suggested. much cleaner now :)